sklearn/utils/estimator_checks.py

```diff
@@ -2486,6 +2488,17 @@ def check_outliers_fit_predict(name, estimator_orig):
             assert_raises(ValueError, estimator.fit_predict, X)
 
 
+def check_fit_non_negative(name, estimator_orig):
+    # Check that proper warning is raised for non-negative X
+    # when tag requires_positive_X is present
+    X = np.array([[-1., 1], [-1., 1]])
+    y = np.array([1, 2])
+    estimator = clone(estimator_orig)
+    assert_raises_regex(ValueError, "Negative values in data passed to",
+                        estimator.fit, X, y)
+
```

Comment:
a redundant link (flake8 error which can't be detected in PR)

Reply:
sorry I don't get what you mean here. My editor does not complain at these lines (but elsewhere in the file)

Reply:
there're 3 lines between this functions and the next function.
E303 too many blank lines (3)

Reply:
my bad... that's what happens when you have so many flake8 errors in the file that you stop paying attention to them :(